configure.ac

```diff
@@ -232,6 +235,47 @@ if test x$ac_cv_func_mkstemp = xyes ; then
 	AC_DEFINE(HAVE_MKSTEMP, 1, [Have mkstemp])
 fi
 
+# std::filesystem::path breaks compilation on OSX Mojave
+# or earlier versions of Xcode, despite the fact it's there
+AC_MSG_CHECKING([if std::filesystem::path compiles fine])
+AC_TRY_COMPILE([
+#if HAVE_FILESYSTEM
+#include <filesystem>
+#endif
+],
+[
+	std::filesystem::path a{};
+],
+	ac_cv_class_fs_path=yes,
+	ac_cv_class_fs_path=no)
+AC_MSG_RESULT($ac_cv_class_fs_path)
+AM_CONDITIONAL([HAVE_FILESYSTEM], [ test "x$ac_cv_class_fs_path" = xyes ])
+if test "x$ac_cv_class_fs_path" = xyes; then
+	AC_DEFINE(USE_STDFILESYSTEM, 1, [Use std::filesystem])
+fi
+
+AC_MSG_CHECKING([if std::filesystem requires linking stdc++fs])
+AC_TRY_LINK([
```

Comment:
AC_TRY_LINK is considered to be obsolete; AC_LINK_IFELSE should be used instead.